x-pack/test/api_integration/deployment_agnostic/apis/observability/dataset_quality/data_stream_details.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogsSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import expect from '@kbn/expect';
+import { log, timerange } from '@kbn/apm-synthtrace-client';
+
+import { SupertestWithRoleScopeType } from '../../../services';
+import { DeploymentAgnosticFtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const roleScopedSupertest = getService('roleScopedSupertest');
+  const synthtrace = getService('synthtrace');
+  const start = '2023-12-11T18:00:00.000Z';
+  const end = '2023-12-11T18:01:00.000Z';
+  const type = 'logs';
+  const dataset = 'nginx.access';
+  const namespace = 'default';
+  const serviceName = 'my-service';
+  const hostName = 'synth-host';
+
+  async function callApiAs(
+    roleScopedSupertestWithCookieCredentials: SupertestWithRoleScopeType,
+    dataStream: string
+  ) {
+    return roleScopedSupertestWithCookieCredentials
+      .get(`/internal/dataset_quality/data_streams/${dataStream}/details`)
+      .query({
+        start,
+        end,
+      });
+  }
+
+  describe('DataStream Details', function () {
+    let synthtraceLogsEsClient: LogsSynthtraceEsClient;
+
+    before(async () => {
+      synthtraceLogsEsClient = await synthtrace.createLogsSynthtraceEsClient();
+
+      await synthtraceLogsEsClient.index([
+        timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) =>
+            log
+              .create()
+              .message('This is a log message')
+              .timestamp(timestamp)
+              .dataset(dataset)
+              .namespace(namespace)
+              .defaults({
+                'log.file.path': '/my-service.log',
+                'service.name': serviceName,
+                'host.name': hostName,
+              })
+          ),
+      ]);
+    });
+
+    after(async () => {
+      await synthtraceLogsEsClient.clean();
+    });
+
+    describe('Viewer User', function () {
+      let supertestViewerWithCookieCredentials: SupertestWithRoleScopeType;
+
+      before(async () => {
+        supertestViewerWithCookieCredentials = await roleScopedSupertest.getSupertestWithRoleScope(
+          'viewer',
+          {
+            useCookieHeader: true,
+            withInternalHeaders: true,
+          }
+        );
+      });
+
+      after(async () => {
+        await supertestViewerWithCookieCredentials.destroy();
+      });
```

Comment:
`destroy` is redundant here. It is only needed when you create instance with API key authentication: `destroy()` removes API key

Cookie header are persistent per role and re-used by different tests within single FTR config